### ISSUE TITLE
Amend Gitter chatroom description to include advice against posting direct answers

### ISSUE THREAD

**Issue by Manish-Giri** [freeCodeCamp/freeCodeCamp#13557]
Having spent quite some time in the Help (and HelpJS) chat rooms, I've seen many instances where people (especially newcomers) post direct answers to campers looking for help with specific challenges. 

I think posting answers is detrimental to the camper's learning process in the long run. Many people just tend to copy-paste the answers posted, instead of understanding the logic involved. Needless to say, they get stuck right on the next challenge and come back asking for help, more confused than ever.

I've also noticed several incidents where people tend to get very offensive if we ask them not to post answers, with replies like "don't tell me what to do" being quite common. Explaining why posting answers is not a good thing, doesn't seem to make much of a difference.

I, therefore, propose that the gitter chatroom(s) description be modified to include a note about not posting direct answers. Just my two cents, though!

Thanks!

**Comment by erictleung** [comment]
@Manish-Giri firstly, thanks for all your help in the Gitter rooms!

I feel from a teaching perspective, just posting answers to "help" others can be frustrating for both parties as you've pointed out. I'm not sure how we can enforce this rule of not posting the answers. As noble as these intentions are to prevent people from finding answers, you can easily Google the answers to many of these challenges. Again, as you've mentioned, people don't seem very keen on being told they can't do particular things in the help rooms. Those help rooms are what I see as very established and probably have their own culture of dealing with helping people.

To really change this posting of direct answers, you'll probably need some more moderators in the rooms to help out. We can also experiment with the idea of actually telling others to not post answers for a given period of time and see how people respond. If the response is positive, we can just keep it. Otherwise, we can evaluate it in a given amount of time.

**tl;dr** I like the idea of not posting answers, but it'll be hard to prevent others from doing so.

I'll leave this to a discussion.

**Comment by coymeetsworld** [comment]
I also spend a good chunk of time in the chatrooms and this is a recurring issue. Usually it's with people that don't know any better and they just want to help. In most cases when you ask nicely to not post direct answers people apologize and change. But others are adamant about not changing, some can even be rude about it. While those cases are rare, they all seem to be of the mindset that if the campers they help don't understand the answer after they're spoon fed to copy and paste, then it's their fault. I don't agree with that logic.

The main problem is that while as Manish said some are rude, they're not really being inappropriate and I'm not sure if it violates any CoC. It would be nice to just have a banner or something when you first come into the room reminding of some general rules rather than having other campers w/ no authority to try and police the situations.

**Comment by coymeetsworld** [comment]
@erictleung we understand we can't stop people from posting direct answers necessarily and we're not asking for more mods either, I'm not sure if that would help much. No one ever said to me "you're not a mod here" or anything like that so the issue isn't authority I don't believe. People mostly tend to be unaware that what they're doing isn't a helpful way of interacting with those who have issues. That's why in most cases there doesn't tend to be huge arguments about this.

As you pointed out if you really wanted to you could find all the answers to challenges and projects on the Internet. People even post "speedruns" for the course which I found absurd. In those cases I would say that is on them and they are making their own bad mistakes. But when other people facilitate that kind of behavior it's something we can act on.

We were more thinking of just having an automated way of notifying chatroom members to help but within reason, something like a motd or camperbot perhaps. This is pending of course whether or not we are in consensus that this is a problem. Manish, myself and others feel it is but it's open to interpretation.

**Comment by coymeetsworld** [comment]
This isn't a request for a challenge, nor is it a bug. This is more of a feature request for gitter.

**Comment by OFFLlNE** [comment]
I think maybe adding a command to camperbot, when calling it would display that posting answers is now allowed (cbot should show the higher "power" status I think)

And those who keep giving answers to people who keep coming back, I just ignore those come-backers and tag the helper to keep helping him.

When one-day they come to ask for help again and there is no answer-giver, then I just tell them that they already should know how to do that and even some very rare cases told to reset to get the idea of the things, he "learned" before

**Comment by texas2010** [comment]
.

**Comment by texas2010** [comment]
I don't know why they didn't realized it is wrong to post answer.  but they know better because teacher told them do not show their answer to other students!  if they do. they are cheat.
I guess they forget about that?

I understand it is very common problem now (my opinion)
I am glad about we could find a way to solve this problem. I hope so.

**Comment by lydatech** [comment]
The issue IMO isn't in the posting of answers, its the posting of answers without any explanation. There are plenty of time where it is appropriate it give someone an answer either due to frustration or what have you but it is imperative that those answers come with an explanation of what the camper was doing wrong and why the solution works the way it does.

Too many times people who are trying to help are flambe' for others in the chat that don't think they should be giving the answers. That isn't the solution either!

Again, IMO the best way to handle helping someone when/if the need arises to give a full answer is to send it to them in a PM, that way it doesn't spoil the challenge for others that don't want to see the answer. But in no way to I think it is up to anyone in chat you flame base the person trying to help. They get rude when asked not to post answers more because of how they are approached by others than having a problem with the request. A little politeness and explanation to the person will go over a lot better that starting to yell at them and try to control their actions.

Thanks everyone for their help in whatever form it may be.

**Comment by lydatech** [comment]
The issue IMO isn't in the posting of answers, its the posting of answers without any explanation. There are plenty of time where it is appropriate it give someone an answer either due to frustration or what have you but it is imperative that those answers come with an explanation of what the camper was doing wrong and why the solution works the way it does.

Too many times people who are trying to help are flambe' for others in the chat that don't think they should be giving the answers. That isn't the solution either!

Again, IMO the best way to handle helping someone when/if the need arises to give a full answer is to send it to them in a PM, that way it doesn't spoil the challenge for others that don't want to see the answer. But in no way to I think it is up to anyone in chat you flame base the person trying to help. They get rude when asked not to post answers more because of how they are approached by others than having a problem with the request. A little politeness and explanation to the person will go over a lot better that starting to yell at them and try to control their actions.

Thanks everyone for their help in whatever form it may be.

**Comment by Manish-Giri** [comment]
> They get rude when asked not to post answers more because of how they are approached by others than having a problem with the request. A little politeness and explanation to the person will go over a lot better that starting to yell at them and try to control their actions.

No one was being rude or impolite or yelling at anyone. All we do is say "please don't post direct answers" with a small explanation of why it's not beneficial. As Coy mentioned earlier, most people take it the right way, while a few people get offended.

Please don't make claims based on facts which aren't 100% accurate.

**Comment by lydatech** [comment]
This wasn't a claim. It was an observation and not directed specifically at you Manish-Giri and it HAS been observed so it isnt an unfounded claim

**Comment by Manish-Giri** [comment]
@lydatech May I know why this issue was closed?

**Comment by lydatech** [comment]
>@lydatech May I know why this issue was closed?
This is more for the forum than the Git Issues. Please direct it there for discussion @Manish-Giri

**Comment by erictleung** [comment]
@Manish-Giri I think @lydatech has a point I didn't think of. Not only is this issue not a bug, but posting in the forum will engage more of the people that are in the chat rooms themselves. So you can definitely have a better and thorough discussion there :+1: But thanks for bringing up the issue. I haven't helped in the Gitter rooms as much as you do but I can see the frustration.
